### PR TITLE
Allow for property name adjustment in the LIMS

### PR DESCRIPTION
**Pull request by KochTobi**
This PR introduces configurable property names for the properties taken from the configured OpenBiS LIMS.
It introduces changes to the assumed property values of the sample status.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,11 +36,11 @@ entry for further detail.
 **Integration with the openBIS LIMS**
 
 The sample information is retrieved automatically from an openBIS instance acting as LIMS. The
-configuration of the openBIS needs to make sure each sample proviedes the following properties:
+configuration of the openBIS needs to make sure each sample provides a QBiC barcode (containin `Q`) and a sample status.
```
Comment (Steffengreiner) [inline README.md]: Nit: Typo 
```suggestion
configuration of the openBIS needs to make sure each sample provides a QBiC barcode (containing `Q`) and a sample status.
```

File: src/main/groovy/life/qbic/samplestatus/reporter/api/LocationService.groovy [hunk 2]
```diff
@@ -18,6 +18,6 @@ interface LocationService {
      * Must not be thrown, if the internal query was successful but no matching location was found. Instead, return an {@link Optional#empty}.
      * @since 0.1.0
      */
-    Optional<Location> getUpdatingPersonLocation() throws ServiceException
+    Optional<Location> getUpdatingPersonLocation()
```
Comment (Steffengreiner) [inline src/main/groovy/life/qbic/samplestatus/reporter/api/LocationService.groovy:21]: Nice spot 👀

File: src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy [hunk 3]
```diff
@@ -32,20 +55,9 @@ class SampleStatusMapper implements Function<String, Result<String, Exception>>
...
-        result = Result.of(SAMPLE_QC_FAIL); break
-      case "LIBRARY_PREP_FINISHED":
-        result = Result.of(LIBRARY_PREP_FINISHED); break
-      default:
-        result = Result.of(new MappingException("Cannot map unkown satus value: $statusString."))
-    }
-    return result
+    return SampleStatus.fromLimsStatus(statusString)
+            .map(it -> Result.of(it.qbicStatus))
+            .orElseGet(() -> Result.of(new MappingException("Cannot map unkown satus value: $statusString.")))
```
Comment (Steffengreiner) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy]: Nit:typo
```suggestion
            .orElseGet(() -> Result.of(new MappingException("Cannot map unknown status value: $statusString.")))
```

Comment (Steffengreiner) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy]: Nice rewrite, however this will always lead to a mappingexception with the currentdata because of: 
1.) Sequencing_complete status
2.) Null and empty values. 
Should this be tackled in a different PR?

**Review by Steffengreiner — DISMISSED** [review]
Heya @KochTobi Nicely done, I found only minor typos and have a question concerning the handling of sample statuses

File: src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy [hunk 4]
```diff
@@ -32,20 +55,9 @@ class SampleStatusMapper implements Function<String, Result<String, Exception>>
     if (statusString.isEmpty()) {
```
Comment (Steffengreiner) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy:58]: Sadly this fails if there is a null value in the database

File: src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy (hunk 3 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/samplestatus/reporter/services/utils/SampleStatusMapper.groovy]: null and empty values are not allowed and thus lead to a failure upon tool execution. This is by design.
As the sequencing complete status cannot be mapped, there is no behaviour we can adopt now. Leading to it being ignored.

**Review by Steffengreiner — APPROVED** [review]
LGTM 🚀
